panther_analysis_tool/main.py

```diff
@@ -489,6 +574,50 @@ def delete_analysis(args: argparse.Namespace) -> Tuple[int, str]:
         return 1, ""
 
     logging.info("Detection(s) %s have been deleted.", " ".join(analysis_id_list))
+    return 0, ""
+
+
+def delete_router(args: argparse.Namespace) -> Tuple[int, str]:
+    # Routes all things delete to the functions they need to go to
+
+    # Get lists of analysis and queries from args
+    analysis_id_list = args.analysis_id
+    query_list = args.query_id
+
+    # If we didn't get a list of analysis, look up analysis associated with queries that were passed
+    if len(analysis_id_list) == 0 and len(query_list) > 0:
+        analysis_id_list = get_analysis_id_by_query(args, query_list)
+
+    # Validate the Detections and queries that we are deleting exist in Panther
+    analysis_id_list = confirm_analysis_exists(args, analysis_id_list)
+    if len(analysis_id_list) == 0 and len(query_list) == 0:
+        logging.error("No matching analysis or queries found, exiting")
+        return 1, ""
+
+    # Similar to above, but we need convert our list of queries into UUIDs so the API understands
+    has_associated_queries = False
+    associated_query_list = get_query_by_analysis_id(args, analysis_id_list)
+    if len(associated_query_list) > 0:
+        has_associated_queries = True
+
+    # Unless explicitly bypassed, get user confirmation to delete
+    if not args.confirm_bypass:
+        analysis_id_string = " ".join(analysis_id_list)
+        logging.warning("You are about to delete detections %s", analysis_id_string)
+        if has_associated_queries:
+            associated_query_string = " ".join(associated_query_list)
+            logging.warning("Scheduled Queries %s will also be deleted", associated_query_string)
+        confirm = input("Continue? (y/n) ")
+
+        if confirm.lower() != "y":
+            print("Cancelled")
+            return 0, ""
+
+    # After confirmation and validation then delete things
+    if has_associated_queries:
+        delete_queries(args, associated_query_list)
```

Comment:
this deletes any queries associated with the analysis list passed in, right? Do we also need to delete any queries passed into the query_list?

Reply:
Logic changed to merge those two lists